palantirnet/search_api_federated_solr
# Umhs 27 restrict

Handles https://palantir.atlassian.net/browse/UMHS-27

This PR depends on the work in https://github.com/palantirnet/federated-search-react/pull/46

###  Testing Instuctions

* In the `federated-search-react` repository, checkout and test branch `UMHS-27-limit-sites` according to the instructions in https://github.com/palantirnet/federated-search-react/pull/46
* In that repo, run `yarn build`
* Copy the new CSS and JS assets to the module directory here.
* Edit `search_api_federated_solr.libraries.yml` to use the new CSS and JS files.
* Clear cache

Now you can test things.

* On the default (non-domain access) Drupal 8 site, add configuration to `settings.php` to enable restricting the list of sites. 

       ```
       $config['search_api_federated_solr.search_app.settings']['facet']['site_name']['site_list'] = [
         'Site name 1',
         'Site name 2',
         'Site name 3',
         'Site name 4',
       ];
       ```

See README.md for more information.

* Run update.php or `drush @fsd-d8.local updb` to install the new schema
* Use Config Inspector to confirm the schema is accurate -- http://d8.fs-demo.local/admin/config/development/configuration/inspect
* Go to configure the app at http://d8.fs-demo.local/admin/config/search-api-federated-solr/search-app/settings
* Note the new field under `Search Results Page > Facets & Filters` which is ` Sites that may be searched from this instance`
* That field should contain the site list that you put in `settings.php.`
* Configure that field to show only two sites.
* Set the default `empty` search to show results.
* Go to http://d8.fs-demo.local/search-app?
* Confirm that only two sites appear in the Site Name filter and that both are selected.
* Confirm that you can use the Site Name filters as expected.
* Confirm that changing the list of allowed sites changes the behavior of search.
* Confirm that setting the list of allowed sites to empty (no selections) show all sites and selects none by default.
* Profit.
